addons/addon-stack-policy/packages/stack-policy/lib/steps/__test__/update-cfn-stack-policy.test.js

```diff
@@ -83,38 +110,236 @@ describe('UpgradeToUserId', () => {
+          }),
+      }));
+      service.cfn.setStackPolicy = jest.fn(() => ({
+        promise: () => Promise.resolve({}),
+      }));
+      const expected = {
+        Statement: [
+          {
+            Effect: 'Allow',
+            Action: 'Update:*',
+            Principal: '*',
+            Resource: '*',
+          },
+          {
+            Effect: 'Deny',
+            Action: 'Update:Delete',
+            Principal: '*',
+            Resource: 'LogicalResourceId/EgressStore*',
+          },
+          {
+            Effect: 'Deny',
+            Action: 'Update:Delete',
+            Principal: '*',
+            Resource: 'LogicalResourceId/AppStream*',
+          },
+        ],
+      };
+
+      await service.execute();
+      expect(service.cfn.getStackPolicy).toHaveBeenCalledTimes(1);
+      expect(service.cfn.setStackPolicy).toHaveBeenCalledWith({
+        StackName: 'backendStackName',
+        StackPolicyBody: JSON.stringify(expected),
+      });
+    });
+
+    it('should successfully update policy when egress is enabled after appstream', async () => {
+      const originalPolicy = {
+        Statement: [
+          {
+            Effect: 'Allow',
+            Action: 'Update:*',
+            Principal: '*',
+            Resource: '*',
+          },
+          {
+            Effect: 'Deny',
+            Action: 'Update:Delete',
+            Principal: '*',
+            Resource: 'LogicalResourceId/AppStream*',
+          },
+        ],
+      };
+      service.cfn.getStackPolicy = jest.fn(() => ({
+        promise: () =>
+          Promise.resolve({
+            StackPolicyBody: JSON.stringify(originalPolicy),
+          }),
+      }));
+      service.cfn.setStackPolicy = jest.fn(() => ({
+        promise: () => Promise.resolve({}),
+      }));
+      const expected = {
+        Statement: [
+          {
+            Effect: 'Allow',
+            Action: 'Update:*',
+            Principal: '*',
+            Resource: '*',
+          },
+          {
+            Effect: 'Deny',
+            Action: 'Update:Delete',
+            Principal: '*',
+            Resource: 'LogicalResourceId/AppStream*',
+          },
+          {
+            Effect: 'Deny',
+            Action: 'Update:Delete',
+            Principal: '*',
+            Resource: 'LogicalResourceId/EgressStore*',
+          },
+        ],
+      };
 
       await service.execute();
       expect(service.cfn.getStackPolicy).toHaveBeenCalledTimes(1);
       expect(service.cfn.setStackPolicy).toHaveBeenCalledWith({
         StackName: 'backendStackName',
-        StackPolicyBody:
-          '{"Statement":[{"Effect":"Allow","Action":"Update:*","Principal":"*","Resource":"*"},{"Effect":"Deny","Action":"Update:Delete","Principal":"*","Resource":"LogicalResourceId/EgressStore*"}]}',
+        StackPolicyBody: JSON.stringify(expected),
       });
     });
+
+    it('should not update policy when no new changes are made', async () => {
```

Comment:
I think we should have a test for the flip scenarios:

1. Disable AppStream if both AppStream and Egress were enabled.
2. Disable AppStream if only AppStream was enabled.

The above can be repeated with AppStream and Egress switched.